packages/iTwinUI-react/src/core/NonIdealState/NonIdealState.tsx

```diff
@@ -0,0 +1,72 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils';
+import cx from 'classnames';
+import '@itwin/itwinui-css/css/non-ideal-state.css';
+
+export type NonIdealStateProps = {
+  /**
+   * An svg component, preferably from @itwin/itwinui-illustrations-react.
+   *
+   * @example
+   * import { Svg404 } from '@itwin/itwinui-illustrations-react';
+   * <NonIdealState svg={<Svg404 />} heading='Not found' />
+   */
+  svg: React.ReactNode;
+
+  /**
+   * Primary heading for the error page
+   */
+  heading?: React.ReactNode;
+
+  /**
+   * Secondary text to explain the error
+   * Can include html in order to provide a hyperlink
```

Comment:
Can you provide an example here like in ErrorPage.tsx?

Reply:
Added example - just copied text from ErrorPage. Should be enough as there is a story with this use case.